liveview-android/src/androidTest/java/com/dockyard/liveviewtest/liveview/components/DatePickerDialogShotTest.kt

```diff
@@ -0,0 +1,284 @@
+package com.dockyard.liveviewtest.liveview.components
+
+import android.icu.text.SimpleDateFormat
+import android.icu.util.Calendar
+import android.icu.util.TimeZone
+import androidx.compose.material3.DatePicker
+import androidx.compose.material3.DatePickerDefaults
+import androidx.compose.material3.DatePickerDialog
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.OutlinedButton
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.material3.rememberDatePickerState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import com.dockyard.liveviewtest.liveview.util.LiveViewComposableTest
+import org.junit.Test
+import java.util.Date
+
+@OptIn(ExperimentalMaterial3Api::class)
+class DatePickerDialogShotTest : LiveViewComposableTest() {
+    private val jun9th2023 = 1686279600000L
+
+    @Test
+    fun simpleDatePickerDialogTest() {
+        val testTag = "myAlert"
+        compareNativeComposableWithTemplate(
+            testTag = testTag,
+            nativeComposable = {
+                DatePickerDialog(
+                    modifier = Modifier.testTag(testTag),
+                    onDismissRequest = {},
+                    confirmButton = {
+                        OutlinedButton(onClick = {}) {
+                            Text(text = "Confirm")
+                        }
+                    },
+                    dismissButton = {
+                        TextButton(onClick = {}) {
+                            Text(text = "Dismiss")
+                        }
+                    },
+                    content = {
+                        DatePicker(state = rememberDatePickerState(initialSelectedDateMillis = jun9th2023))
+                    }
+                )
+            },
+            template = """
+                <DatePickerDialog phx-click="hideDialog" test-tag="$testTag">
+                  <OutlinedButton phx-click="hideDialog" template="confirm">
```

Comment:
is `template` a Jetpack reserved word?

Reply:
In general I don't know. In this case, it's just a named argument to [`compareNativeComposableWithTemplate`](https://github.com/liveview-native/liveview-client-jetpack/blob/0631b311fd152b91ab91f4422caecee60ac3e899/liveview-android/src/androidTest/java/com/dockyard/liveviewtest/liveview/util/LiveViewComposableTest.kt#L19-L29).

Reply:
If it is part of the Jetpack API then this will impact how we've done things in the SwiftUII client. `template` has a meaning that is specific to how we render out named content sections in SwiftUI. Granted that method may not apply to Jetpack but I'd like to avoid any confusion around the use of `template` between the frameworks. I believe we may need to namespace the attribute in the SwiftUI client. In fact, we should probably always be namespacing whenever we run into requiring the use of an attribute that isn't part of the UI framework's API but we still need it.

Pending @nelson-glauber's feedback on this attr, but I'll work with @carson-katri to change this in the swiftui client for v0.3

Reply:
No. This approach is being used to identify one specific child of a particular component.